clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java

```diff
@@ -247,7 +249,31 @@ public EhcacheEntityResponse invoke(ClientDescriptor clientDescriptor, EhcacheEn
 
   @Override
   public void handleReconnect(ClientDescriptor clientDescriptor, byte[] extendedReconnectData) {
-    //nothing to do
+    ClientState clientState = this.clientStateMap.get(clientDescriptor);
+    if (clientState == null) {
+      LOGGER.error("Client {} trying to reconnect is not connected to entity.", clientDescriptor);
+      return;
```

Comment:
I would throw an AssertionError here - this must be impossible from the platform's perspective!